### PR TITLE
UI shell header submenu

### LINKED ISSUE #1355: Create Header component
This issue is for implementing the header component found in:

image

Latest sketch file: https://ibm.box.com/s/4e4zrz6mlo0jpr8lw4vimwlxn4vc59cy

### PR DESCRIPTION
**Pull request by emyarod**
Closes #1355 

This PR will implement header submenu logic for the UI Shell

#### Changelog

**New**

- header submenu opening/closing logic
- keyboard navigation for header submenus

### CODE REVIEW

**Review by asudoh — APPROVED** [review]
LGTM :+1::tada: - Thanks @emyarod for completing this!!

**Review by joshblack — CHANGES_REQUESTED** [review]
Looks amazing so far! 🎉 Thanks for adding this in 🙏 Couple of things I noticed when going through it:

- The menus should display on hover versus click (I think in React we used mouseover/mouseleave)
- Only one menu item in the bar should have `tabindex="0"`, the rest should be `-1`. By default the first item has this set. After interacting with the widget, the most recently focused nav item should have it set.
- When pressing arrow down on a menu, it should move focus to the first item in the list
- When inside a menu, using arrow left or right should close the menu and shift focus to the previous/next nav item
- Home/End should be added for navigating to the first/last item
- As a form of enhancement, when focusing on a menu item typing a letter should jump to the next nav item that first matches that letter

**Comment by joshblack** [comment]
I think in our README we have some of the examples that can help depict what I think the behavior is that we want to go for keyboard-wise: https://github.com/IBM/carbon-components/tree/8bd1cee785021cdb1e3fc8010f897e2082bd3cb2/src/components/ui-shell#header

**Comment by joshblack** [comment]
@emyarod could you add a staging link and add @aagonzales as a reviewer btw?

**Comment by emyarod** [comment]
staging link: http://ui-shell-header-submenu-delightful-parrot.mybluemix.net/component/ui-shell--default.html

**Review by aagonzales — COMMENTED** [review]
Looks good!! Just want to confirm something. For the `bx--header_action` in the top right when you click on them it goes blue. The blue fill should be the selected state and the on click color should be #3D3D3D/Gray 80. 

And one small design change, any time there's a hover can we have the hovered text change to White?

**Comment by emyarod** [comment]
thanks for the feedback @aagonzales! sorry there was a bit of guesswork with the styles since I had to make some assumptions based on the Sketch file I have. Just want to double check something:

> For the `bx--header_action` in the top right when you click on them it goes blue. The blue fill should be the selected state and the on click color should be #3D3D3D/Gray 80.

Right now the `bx--header__action` is also `#3d3d3d` on hover. Do we want the on click color to be identical to the hover color?

**Comment by aagonzales** [comment]
For `bx--header__action` hover should be `#333333` and active should be `#3d3d3d`

File: src/components/ui-shell/_header.scss [hunk 1]
```diff
@@ -30,12 +30,11 @@
   }
 
   .#{$prefix}--header__action:hover {
-    background-color: $shell-ui-02;
+    background-color: #333;
```
Comment (alisonjoseph) [inline src/components/ui-shell/_header.scss]: this should probably be a token

**Review by aagonzales — APPROVED** [review]

**Review by marijohannessen — APPROVED** [review]
It looks good to me! 👍 ✅

**Review by alisonjoseph — APPROVED** [review]

**Pull request merged**